File: JavaScript/7js 3/skiponacci.js
/*
|--------------------------------------------------------------------------
| The Skiponacci Sequence
|--------------------------------------------------------------------------
|
*/

function skiponacci(n) {
   let res = [];
   let a = 1;
   let b = 1;
   for (let i = 0; i < n; i++) {
      res.push(i % 2 == 0 ? a : "skip");
      i % 2 == 0 ? (a = a + b) : (b = a + b);
   }
   return res.join(" ");
}


function skiponacci(n, a = 1, b = 0, arr = []) {
   if (!n) return arr.map((i, ind) => (ind & 1 ? "skip" : i)).join` `;
   return skiponacci(n - 1, b, a + b, [...arr, a + b]);
}


function skiponacci(n) {
   let a = 0;
   let b = 1;
   return Array.from({ length: n }, (_, i) => {
      [a, b] = [b, a + b];
      return i & 1 ? "skip" : a;
   }).join(" ");
}


console.log(skiponacci(1), "1");
console.log(skiponacci(5), "1 skip 2 skip 5");
console.log(skiponacci(7), "1 skip 2 skip 5 skip 13");

File: Python/7py 3/skiponacci.py
"""
|--------------------------------------------------------------------------
| The Skiponacci Sequence
|--------------------------------------------------------------------------
|
"""

def skiponacci(n):
   a, b, res = 1, 1, []
   for i in range(n):
      res.append(str(a) if i % 2 == 0 else "skip")
      if i % 2 == 0: a = a + b
      else: b = a + b
   return " ".join(res)


def skiponacci(n):
   fib = [1, 1][:n]
   for _ in range(n-2): fib.append(sum(fib[-2:]))
   return " ".join(str(n) if i % 2 else "skip" for i, n in enumerate(fib, 1))


def skiponacci(n):
   res = []
   a, b = 0, 1
   for i in range(n):
      a, b = a + b, a
      res.append('skip' if i%2 else str(a))
   return ' '.join(res)


print(skiponacci(1), "1")
print(skiponacci(5), "1 skip 2 skip 5")
print(skiponacci(7), "1 skip 2 skip 5 skip 13")